src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,203 @@
+     * @throws DukeException A custom Exception that carries a message for the user if thrown.
+     */
+    public String processCommand(String command) throws DukeException {
+        Pattern pattern = Pattern.compile("^(.*?)\\s(.*?)(?:\\s/..\\s?(.*))?$");
+        Matcher matcher = pattern.matcher(command);
+        if (command.equals("list")) {
+            return this.list.toString();
+        }
+        if (command.equals("hello")) {
+            return "Hi! I'm Duke! Pleasure to meet you :)";
+        }
+        if (command.equals("bye")) {
+            this.isExit = true;
+            Storage storage = new Storage();
+            storage.updateFile(this.list);
+            return "Bye! Hope to see you again soon!";
+        }
+        if (command.equals("help")) {
+            return "Accepted commands:\n"
+                    + "hello - hello!\n"
+                    + "list - show current list\n"
+                    + "bye - saves the current list and exits the program\n"
+                    + "\n"
+                    + "todo <description> - create a todo Task\n"
+                    + "event <description> /at <dd/MM/yyyy> - create an event Task (date is optional)\n"
+                    + "deadline <description> /by <dd/MM/yyyy> - create a deadline Task (date is optional)\n"
+                    + "\n"
+                    + "done <index> - mark the specified task as done\n"
+                    + "undo <index> - mark the specified task as not done\n"
+                    + "delete <index> - deletes the specified task from the list";
+        }
+        if (matcher.find()) {
+            String com = matcher.group(1);
+            String task = matcher.group(2);
+            String date = matcher.group(3);
+            String index = matcher.group(2);
+
+            switch (com) {
+            case ("undo"):
+            case ("done"):
+            case ("delete"):
+            case("find"):
+                return this.processList(com, index);
+
+            case ("todo"):
+            case ("deadline"):
+            case ("event"):
+                return this.processTask(com, task, date);
+            default:
+                throw new DukeException("Sorry, I did not understand: " + command
+                        + ".\nUse \"help\" to look at available commands.");
+            }
+        }
+        throw new DukeException("Sorry, I did not understand: " + command
+                + ".\nUse \"help\" to look at available commands.");
+    }
+
+    /**
+     * Helper function to process Task related commands.
+     *
+     * @param com The type of Task.
+     * @param task The description of the Task.
+     * @param date The date of the Task to be done at or by (if applicable).
+     * @return String result of the function executed.
+     * @throws DukeException A custom Exception that carries a message for the user if thrown.
+     */
+    private String processTask(String com, String task, String date) throws DukeException {
```

Comment:
Hi, I think the naming of your paramaters could be made more clearer, such that com --> command. Sometimes it is just confusing to read a contracted or shortened word.